plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerDelegate.java

```diff
@@ -133,6 +133,10 @@ IStatus checkConflictingServiceIds(IModule[] current, IModule[] toBeAdded,
     }
     if (toBeAdded != null) {
       for (IModule module : toBeAdded) {
+        if (currentServiceIds.containsValue(module)) {
```

Comment:
what's the equals() implementation on a module? The docs are explicitly unclear about this  http://www.eclipse.org/webtools/wst/api/org/eclipse/wst/server/core/IModule.html but if it's just checking the IDs, this would seem to render the entire loop moot.